_posts/2023-02-22-#25038.md

```diff
@@ -7,18 +7,258 @@ authors: [glozow]
+- RBF carve out is not needed.
+- Given an unconfirmed v3 transaction of size `s` paying fees `f`, a maximum-size child paying the same feerate pays  `f/s * 1000` in fees.
+
+Further work that may enable more benefits:
+- The [Ephemeral Outputs](https://github.com/bitcoin/bitcoin/pull/26403) proposal builds on top of v3, offering an
+alternative to [SIGHASH\_GROUP](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2021-July/019243.html), with
+most of the features and a policy-only changeset.
+- Ephemeral outputs may be used to enable [Eltoo](https://bitcoinops.org/en/topics/eltoo/), which also
+[hinges](https://bitcoinops.org/en/newsletters/2022/05/18/#using-transaction-introspection-to-prevent-rbf-pinning) on the resolution of these RBF limitations.
+- As v3 and package RBF would allow counterparties to replace each others' commitment transactions, just a single anchor
+  output per commitment transaction could be sufficient. This may eventually allow the removal of
+[CPFP carve out](https://github.com/bitcoin/bitcoin/blob/master/doc/policy/mempool-limits.md#cpfp-carve-out).
+
 ## Questions
-1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
 
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+What was your review approach?
+
+### Concept
+
+1. What is the difference between a transaction that "explicitly" signals BIP125 and one that "inherits" BIP125
+   signaling? Why is it impossible for a v3 transaction to be one but not the other?
+
+1. Why should there be a rule to check whether a replacement transation is more incentive compatible to mine?
+
+1. Is min(package feerate, individual feerate) an accurate incentive compatibility score for v3 transactions? What
+   about in general? (See `CheckMinerScores()` [implementation](https://github.com/bitcoin-core-review-club/bitcoin/commit/58e811ecb1e1977422ecda2af62460e8efc057be)).
+
+1. Imagine that we *really* need a transaction txA to confirm immediately, but a malicious counterparty has broadcast a conflicting
+   transaction txB which is 2000vB, paying 10,000sat in fees (5sat/vB). Let's say we are ok with either txA or txB
+confirming, as long as it's in the next block, and it's guaranteed (just pretend, ok?) that anything above 20sat/vB will
+be mined immediately, while anything below 20sat/vB will not.  What fee can we put on our transaction to *guarantee*
+that either txA or txB will reach a package feerate of 20sat/vB? (Hint: txA needs to replace txB and any descendants it
+may have.  What is the maximum fee the counterparty can attach to descendants of txB without bumping it to 20sat/vB?)
```

Comment:
I'm quite confused by this question. First you say that 
> Imagine that we *really* need a transaction txA to confirm immediately

but then next sentence is
> Let's say we are ok with either txA or txB confirming

Those seem conflicting?

And then
> What fee can we put on our transaction to *guarantee* that either txA or txB will reach a package feerate of 20sat/vB?

seems to steer towards TxB affecting the package feerate of TxA, but since this (how I read the question) is RBF and not CPFP I think that's confusing? 

Finally - I think it needs specifying if this is in a v2 or v3 world?

Reply:
Clarified